base/shell/explorer/appbar.cpp

```diff
@@ -198,22 +200,33 @@ static UINT_PTR handle_appbarmessage(DWORD msg, _AppBarData *abd)
         }
         return TRUE;
     case ABM_GETSTATE:
-        FIXME("SHAppBarMessage(ABM_GETSTATE): stub\n");
-        return ABS_ALWAYSONTOP | ABS_AUTOHIDE;
+        TRACE("SHAppBarMessage(ABM_GETSTATE)\n");
+        return (g_TaskbarSettings.sr.AutoHide ? ABS_AUTOHIDE : 0) |
+               (g_TaskbarSettings.sr.AlwaysOnTop ? ABS_ALWAYSONTOP : 0);
+    case ABM_SETSTATE:
+        TRACE("SHAppBarMessage(ABM_SETSTATE lparam=%s)\n", wine_dbgstr_longlong(abd->lParam));
+        hwnd = GetPrimaryTaskbar();
+        if (hwnd)
+        {
+            TaskbarSettings settings = g_TaskbarSettings;
+            settings.sr.AutoHide = (abd->lParam & ABS_AUTOHIDE) != 0;
+            settings.sr.AlwaysOnTop = (abd->lParam & ABS_ALWAYSONTOP) != 0;
+            SendMessageW(hwnd, TWM_SETTINGSCHANGED, 0, (LPARAM)&settings);
+            return TRUE;
+        }
+        return FALSE;
     case ABM_GETTASKBARPOS:
-        FIXME("SHAppBarMessage(ABM_GETTASKBARPOS, hwnd=%p): stub\n", hwnd);
-        /* Report the taskbar is at the bottom of the screen. */
-        abd->rc.left = 0;
-        abd->rc.right = GetSystemMetrics(SM_CXSCREEN);
-        abd->rc.bottom = GetSystemMetrics(SM_CYSCREEN);
-        abd->rc.top = abd->rc.bottom-1;
-        abd->uEdge = ABE_BOTTOM;
-        return TRUE;
+        TRACE("SHAppBarMessage(ABM_GETTASKBARPOS, hwnd=%p)\n", hwnd);
+        abd->uEdge = g_TaskbarSettings.sr.Position;
+        abd->hWnd = GetPrimaryTaskbar();
```

Comment:
Does this message only apply to the "primary" taskbar (whatever that is)?
(and same question for the two others that use GetPrimaryTaskbar)

Reply:
Everything taskbar related applies to the "not secondary taskbar" AKA the taskbar on the primary monitor. 

For ABM_GETAUTOHIDEBAR MSDN says:

 > If the system has multiple monitors, the monitor that contains the primary taskbar is used.

ABM_GETAUTOHIDEBAREX & ABM_SETAUTOHIDEBAREX are the only multi-mon specific messages.